docs/new-editor.md

```diff
@@ -0,0 +1,163 @@
+# Integrating a new editor
+
+Before writing a new editor client, first check if someone else has managed to
+integrate metals with your favorite text editor.
+
+* [VS Code](https://github.com/scalameta/metals/blob/master/vscode-extension/src/extension.ts),
```

Comment:
Since this is an "official" list, s/VS Code/Visual Studio Code ? It's fine to abbreviate in the rest of the document

Reply:
Done.